### PR TITLE
Add travis config '.travis.yml'

### PR DESCRIPTION
**Pull request by wkentaro**

### CODE REVIEW

File: pcl_ros/CMakeLists.txt [hunk 1]
```diff
@@ -4,7 +4,6 @@ project(pcl_ros)
 ## Find system dependencies
 find_package(cmake_modules REQUIRED)
 find_package(Boost REQUIRED COMPONENTS system filesystem thread)
-find_package(Eigen3 REQUIRED)
```
Comment (wkentaro) [inline pcl_ros/CMakeLists.txt]: Without this commit, it raises not found error.

Comment (paulbovbel) [inline pcl_ros/CMakeLists.txt]: Please remove these changes. Eigen is very much needed for pcl_ros, and I don't understand why you're asking to remove it. Provide more context in the PR description.

**Comment by wkentaro** [comment]
```
CMake Error at CMakeLists.txt:7 (find_package):
  By not providing "FindEigen3.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Eigen3", but
  CMake did not find one.
  Could not find a package configuration file provided by "Eigen3" with any
  of the following names:
    Eigen3Config.cmake
    eigen3-config.cmake
  Add the installation prefix of "Eigen3" to CMAKE_PREFIX_PATH or set
  "Eigen3_DIR" to a directory containing one of the above files.  If "Eigen3"
  provides a separate development package or SDK, be sure it has been
  installed.
```

**Comment by paulbovbel** [comment]
You cannot remove Eigen3 from this package. I'm assuming you did this for a reason, like if travis reports an error with the cmake module, but you have provided no context in your PR description.

**Comment by wkentaro** [comment]
Sorry for fiew descriptions.
I will make it understood.

**Comment by wkentaro** [comment]
The reason I removed the find_package of Eigen3 is to fix this error 

```
CMake Error at CMakeLists.txt:7 (find_package):
  By not providing "FindEigen3.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Eigen3", but
  CMake did not find one.
  Could not find a package configuration file provided by "Eigen3" with any
  of the following names:
    Eigen3Config.cmake
    eigen3-config.cmake
  Add the installation prefix of "Eigen3" to CMAKE_PREFIX_PATH or set
  "Eigen3_DIR" to a directory containing one of the above files.  If "Eigen3"
  provides a separate development package or SDK, be sure it has been
  installed.
```

as shown in https://s3.amazonaws.com/archive.travis-ci.org/jobs/120628283/log.txt .
Removing find_package Eigen3 fixed the problem on travis, and this seems related to https://github.com/ros-perception/perception_pcl/issues/106 https://github.com/ros-perception/perception_pcl/pull/108 issues.

**Comment by paulbovbel** [comment]
It is related to those issues - thanks for putting up a travis config, I'll try to investigate how to get it working with an Eigen cmake module

**Comment by wkentaro** [comment]
For debian:stretch, qtbase5-dev is required. for `find_package(Qt5Widgets QUIET)`.
```
$ dpkg -S /usr/lib/x86_64-linux-gnu/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake
qtbase5-dev:amd64: /usr/lib/x86_64-linux-gnu/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake
```

But by default, it is not installed and the test fails:

```
[1m[31mErrors[0m    [31m <<[0m [36mpcl_ros[0m:[34mmake[0m [1m[31m/home/travis/catkin_ws/logs/pcl_ros/build.make.000.log[0m[0m
/usr/bin/ld: cannot find -lQt5::Widgets

collect2: error: ld returned 1 exit status

make[2]: *** [/home/travis/catkin_ws/devel/.private/pcl_ros/lib/pcl_ros/convert_pcd_to_image] Error 1

make[1]: *** [CMakeFiles/convert_pcd_to_image.dir/all] Error 2

make: *** [all] Error 2
```

So, `qtbase5-dev` needs to be added to package.xml

```
root@1827d2d78f77:~/catkin_ws/src# rosdep db --os debian:stretch | grep qtbase5-dev
qtbase5-dev -> qtbase5-dev
```

**Comment by wkentaro** [comment]
@paulbovbel Actually I need this before merging https://github.com/ros-perception/perception_pcl/pull/118, so could you please review this next?

**Comment by paulbovbel** [comment]
Just so I understand, what does travis give us the buildfarm PR tests do not?

**Comment by wkentaro** [comment]
It tests PRs before merging it.

**Comment by paulbovbel** [comment]
Right, but buildfarm does the same thing in a standard way, just need to enable it in https://github.com/ros/rosdistro/blob/master/lunar/distribution.yaml#L1178 via:

```
    source:
      test_pull_requests: true
```

**Comment by wkentaro** [comment]
Oh, I didn't know the way. I will try that.

**Comment by mikaelarguedas** [comment]
@paulbovbel @wkentaro Yes the ROS buildfarm allows you to test your pull requests before merging. The only difference with this travis setup is that the buildfarm will test only for the targeted Ubuntu LTS. So for Lunar it will test only on Ubuntu Xenial (in other words the issue we faced about qt5 on jessie would not have been caught by the buildferm pull request testing).

**Comment by wkentaro** [comment]
@mikaelarguedas Thank you for the information!
Well then, it seems that Travis is better than the ROS buildfarm for me. What do you think? @paulbovbel

**Review by paulbovbel — APPROVED** [review]

**Comment by paulbovbel** [comment]
Sounds good!

**Pull request merged**